debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlAntlrDdlParser.java

```diff
@@ -213,15 +213,39 @@ public String parseName(MySqlParser.UidContext uidContext) {
      * @return qualified {@link TableId}.
      */
     public TableId parseQualifiedTableId(MySqlParser.FullIdContext fullIdContext) {
-        String fullTableName = fullIdContext.getText();
-        int dotIndex;
-        if ((dotIndex = fullTableName.indexOf(".")) > 0) {
-            return resolveTableId(withoutQuotes(fullTableName.substring(0, dotIndex)),
-                    withoutQuotes(fullTableName.substring(dotIndex + 1, fullTableName.length())));
-        }
-        else {
-            return resolveTableId(currentSchema(), withoutQuotes(fullTableName));
+        final char[] fullTableName = fullIdContext.getText().toCharArray();
+        StringBuilder component = new StringBuilder();
+        String dbName = null;
+        String tableName = null;
+        final char EMPTY = '\0';
+        char lastQuote = EMPTY;
+        for (char c : fullTableName) {
+            if (isQuote(c)) {
+                // Opening quote
+                if (lastQuote == EMPTY) {
+                    lastQuote = c;
+                }
+                // Closing quote
+                else if (lastQuote == c) {
+                    lastQuote = EMPTY;
+                }
+                // Quote that is part of name
```

Comment:
Do we have any test for this case actually? How would escaping of the quote look like?

Reply:
@gunnarmorling Added more comprehensive test and escaping support